backend/stakepoold/server.go

```diff
@@ -301,46 +484,297 @@ func main() {
+		return nil
+	}
+
+	return nil
+}
+
+// saveData saves some appContext fields to a file so they can be loaded back
+// into memory at next run.
+func saveData(ctx *appContext) {
+	ctx.mtx.Lock()
+	defer ctx.mtx.Unlock()
+
+	saveFiles := getDataNames()
+
+	for filenameprefix, dataversion := range saveFiles {
+		t := time.Now()
+		destFilename := strings.Replace(dataFilenameTemplate, "KIND", filenameprefix, -1)
+		destFilename = strings.Replace(destFilename, "DATE", t.Format("2006_01_02_15_04_05"), -1)
+		destFilename = strings.Replace(destFilename, "VERSION", dataversion, -1)
+		destPath := strings.ToLower(filepath.Join(ctx.dataPath, destFilename))
+
+		// Pre-validate whether we'll be saving or not.
+		switch filenameprefix {
+		case "LiveTickets":
+			if len(ctx.liveTicketsMSA) == 0 {
+				log.Warn("saveData: liveTicketsMSA is empty; skipping save")
+				continue
+			}
+		case "ManuallyAddedTickets":
+			if len(ctx.manuallyAddedTicketsMSA) == 0 {
+				// we don't expect to do anything with this yet so don't warn
+				// log.Warn("saveData: manuallyAddedTicketsMSA is empty; skipping save")
+				continue
+			}
+		case "Stats":
+			// Always save stats
+			break
+		case "UserVotingConfig":
+			if len(ctx.userVotingConfig) == 0 {
+				log.Warn("saveData: UserVotingConfig is empty; skipping save")
+				continue
+			}
+		}
+
+		w, err := os.Create(destPath)
+		defer w.Close()
+		if err != nil {
+			log.Errorf("Error opening file %s: %v", ctx.dataPath, err)
+			continue
+		}
+
+		switch filenameprefix {
+		case "LiveTickets":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.liveTicketsMSA); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "ManuallyAddedTickets":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.manuallyAddedTicketsMSA); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "Stats":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.stats); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "UserVotingConfig":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.userVotingConfig); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		}
+
+		log.Infof("saveData: successfully saved %v data to %s",
+			filenameprefix, destPath)
+	}
+
 }
 
-// winner contains all the bits and pieces required to vote and to print
-// statistics after usage.
-type winner struct {
+// ticketMetadata contains all the bits and pieces required to vote new tickets,
+// to look up new/missed/spent tickets, and to print statistics after usage.
+type ticketMetadata struct {
+	blockHeight  int64
 	msa          string                    // multisig
 	ticket       *chainhash.Hash           // ticket
+	spent        bool                      // spent (true) or missed (false)
```

Comment:
Yeah I just copied how it was named in the notification. I need to look at how/where that is set a bit more closely I guess.